### PR TITLE
Spanish translation ch10-00-generics.md

### PR DESCRIPTION
**Pull request by jossue33**
Number of words: 928

### CODE REVIEW

File: second-edition/src/ch10-00-generics.md [hunk 1]
```diff
@@ -1,49 +1,49 @@
...
-that’s fresh in our minds, we’ll use the same mechanics with generics to
-extract a generic function! In the same way that you recognize duplicated code
-to extract into a function, you’ll start to recognize duplicated code that can
-use generics.
-
-Consider a small program that finds the largest number in a list, shown in
-Listing 10-1:
-
-<span class="filename">Filename: src/main.rs</span>
+# Tipos de genéricos, rasgos y tiempos de vida
```
Comment (thecodix) [inline second-edition/src/ch10-00-generics.md]: Tipos Genéricos, Rasgos y Tiempos de Vida

File: second-edition/src/ch10-00-generics.md [hunk 2]
```diff
@@ -1,49 +1,49 @@
...
+manera genérica. Los rasgos pueden ser combinados con tipos genéricos para limitar un
+tipo generico para esos tipos que tienen un comportamiento particular, en vez de 
+cualquier otro tipo.
+
+Finalmente, discutiremos los *los tiempos de vida*, los cuales son un tipo de genérico que nos deja
+darle al compilador información sobre cómo las referencias están relacionadas entre sí
+Los tiempos de vida son las característica en Rust que nos deja tomar prestados valores en varias 
+situaciones y tener la prueba del compilador de que las referencias serán válidas.
+
+## Retirando la duplicación al extraer una función
```
Comment (thecodix) [inline second-edition/src/ch10-00-generics.md]: Evitar la duplicación

File: second-edition/src/ch10-00-generics.md [hunk 3]
```diff
@@ -158,31 +158,31 @@ fn main() {
...
-different ways in different scenarios. In the same way that the function body
-is now operating on an abstract `list` instead of concrete values, code using
-generics will operate on abstract types. The concepts powering generics are the
-same concepts you already know that power functions, just applied in different
-ways.
-
-What if we had two functions, one that found the largest item in a slice of
-`i32` values and one that found the largest item in a slice of `char` values?
-How would we get rid of that duplication? Let’s find out!
+<span class="caption">Listado 10-3: Código de abstracción para encontrar el número más grande
```
Comment (thecodix) [inline second-edition/src/ch10-00-generics.md]: Código abstracto

**Comment by thecodix** [comment]
Left some comments, if you're so kind to work on it :)

**Comment by jossue33** [comment]
@thecodix todo listo.

**Comment by thecodix** [comment]
Perfect

**Comment by thecodix** [comment]
machine translated

**Pull request merged**